### PR TITLE
Remove GoogleAnalytics specific logic from the FE

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This is the FE-facing part of the #42792

### CODE REVIEW

File: frontend/src/metabase-lib/v1/queries/utils/segments.js [hunk 1]
```diff
@@ -9,21 +9,8 @@ export function getSegmentOrMetricQuestion(query, table, metadata) {
...
-        .question();
-    }
-  } else {
-    return question
-      .legacyQuery({ useStructuredQuery: true })
-      .aggregate(["count"])
-      .question();
-  }
-
-  return null;
```
Comment (rafpaf) [inline frontend/src/metabase-lib/v1/queries/utils/segments.js:28]: 🤔

**Review by rafpaf — APPROVED** [review]

File: frontend/src/metabase-lib/v1/queries/utils/segments.js (hunk 1 above)
Comment (rafpaf) [reply frontend/src/metabase-lib/v1/queries/utils/segments.js:28]: Just puzzled why we were ever returning null here...

**Review by ranquild — APPROVED** [review]

**Comment by ranquild** [comment]
@nemanjaglumac backport to 0.50?

File: frontend/src/metabase-lib/v1/queries/utils/segments.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/src/metabase-lib/v1/queries/utils/segments.js:28]: Oooooooor

if entity type was `GoogleAnalyticsTable`, but NOT `dateField` => return `null`?

Weird placement, nevertheless.

Comment (rafpaf) [reply frontend/src/metabase-lib/v1/queries/utils/segments.js:28]: Right, I missed that

**Pull request merged**